clang/include/clang/Basic/DiagnosticASTKinds.td

```diff
@@ -1016,4 +1016,7 @@ def warn_unpacked_field
 def warn_unaligned_access : Warning<
   "field %1 within %0 is less aligned than %2 and is usually due to %0 being "
   "packed, which can lead to unaligned accesses">, InGroup<UnalignedAccess>, DefaultIgnore;
+
+def err_cannot_mangle_expression : Error<
+  "cannot yet mangle %0 expression">;
```

Comment:
All of the other "cannot yet mangle" diagnostics are done with a custom diagnostic, as in https://github.com/llvm/llvm-project/blob/475e154331af19f175ec082b08547b155bba1577/clang/lib/AST/ItaniumMangle.cpp#L5116 -- I think we can drop this diagnostic and do the same (that helps make it more obvious in the code that the diagnostic is really a fixme issue rather than a diagnostic we intend to keep long-term). WDYT?

Reply:
I think a `// FIXME` would make that a lot more obvious than using the custom diagnostics engine. To me it wasn't obvious that these should be FIXMES, which is the reason I replaced them with a normal diagnostic.

Reply:
Ah, so the "obvious" may be more obvious to people used to working with Clang's diagnostics and not so obvious to folks not as well-versed. Good feedback!

I'm okay with the changes but I have a preference for doing this in two steps. Step 1: your patch w/custom diagnostics engine, Step 2: replace all custom diagnostics with the regular diagnostic in one go. (This makes code archeology easier because the changes are kind of orthogonal to one another.) Would you be okay with splitting it up despite the extra work?

Reply:
Sounds good!